### PR TITLE
Aflplusplus

### PR DESCRIPTION
**Pull request by devnexen**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/aflplusplus.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class Aflplusplus < Formula
+  desc "American fuzzy lop: Security-oriented fuzzer"
+  homepage "https://github.com/vanhauser-thc/AFLplusplus"
+  url "https://github.com/vanhauser-thc/AFLplusplus/archive/2.62c.tar.gz"
+  sha256 "cde181ac733aa3a1212ffcb494bb9306a2086c7521fb006719b0e15cd8015c63"
+
+  depends_on "automake"
```
Comment (SMillerDev) [inline Formula/aflplusplus.rb]: Is this a runtime dependency?

File: Formula/aflplusplus.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+class Aflplusplus < Formula
+  desc "American fuzzy lop: Security-oriented fuzzer"
+  homepage "https://github.com/vanhauser-thc/AFLplusplus"
+  url "https://github.com/vanhauser-thc/AFLplusplus/archive/2.62c.tar.gz"
+  sha256 "cde181ac733aa3a1212ffcb494bb9306a2086c7521fb006719b0e15cd8015c63"
+
+  depends_on "automake" => :build
+  depends_on "wget" => :build
```
Comment (SMillerDev) [inline Formula/aflplusplus.rb]: Why does it need wget only on build time?

Comment (devnexen) [reply Formula/aflplusplus.rb]: there is a unicorn for AFL release to download during the build process.

Comment (SMillerDev) [reply Formula/aflplusplus.rb]: Make scripts shouldn't download anything when building. If you need a file you should use a resource block.

Comment (devnexen) [reply Formula/aflplusplus.rb]: Surely ... but the shell script checks the presence of wget anyway.

Comment (SMillerDev) [reply Formula/aflplusplus.rb]: That seems worth patching upstream.

Comment (devnexen) [reply Formula/aflplusplus.rb]: done it ... is cloning GitHub repo doable in a build ?

Comment (Bo98) [reply Formula/aflplusplus.rb]: Yes, but in this case you probably want to be applying a patch instead (using a `patch do` block with https://github.com/vanhauser-thc/AFLplusplus/commit/9ddd7e0e3f6f9dc8b837e96b508db7fd1f9a1a19.patch?full_index=1). When cloning a repo we would still want to follow the tag.

Comment (devnexen) [reply Formula/aflplusplus.rb]: Finally opted for building the essential bit, might need further upstream changes.

**Pull request closed without merging**